### PR TITLE
Add google protos to buildbucket update version, deps

### PR DESCRIPTION
**Pull request by ricardoamador**
A couple of changes added for buildbucket support:
* Add the supporting well known Google protos so buildbucket objects can be instantiated.
* Add short instructions on developing changes locally before pushing the library to pub.dev.
* Re-pin versions as per: https://github.com/flutter/flutter/issues/110272

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/buildbucket-dart/pubspec.yaml
Comment (keyonghan) [inline packages/buildbucket-dart/pubspec.yaml:13]: Copied earlier comment here: https://github.com/flutter/cocoon/pull/3492/files#r1491771510

> Any reason to use the undetermined caret version? We have been trying to use fixed versions to avoid potential breakage.

Can we try finding a compatible combination?

**Review by keyonghan — COMMENTED** [review]
Do you want to revert https://github.com/flutter/cocoon/pull/3492 or rebase/make changes on top of https://github.com/flutter/cocoon/pull/3492?

File: packages/buildbucket-dart/pubspec.yaml
Comment (keyonghan) [reply packages/buildbucket-dart/pubspec.yaml:13]: Let's pin to specific versions as other top level folders. More context why we do this way: https://github.com/flutter/flutter/issues/110272

**Comment by ricardoamador** [comment]
> Do you want to revert #3492 or rebase/make changes on top of #3492?

I can revert it. Let me do some clean up and take care of this today.

File: packages/buildbucket-dart/pubspec.yaml
Comment (ricardoamador) [reply packages/buildbucket-dart/pubspec.yaml:13]: Okay no problem! Thank you for the context on this!

File: packages/buildbucket-dart/CHANGELOG.md
Comment (keyonghan) [inline packages/buildbucket-dart/CHANGELOG.md:7]: Shall we release one version each time by combining these two changes?

Comment (ricardoamador) [reply packages/buildbucket-dart/CHANGELOG.md:7]: So I had already deployed 1.0.8 by mistake. Which is why I updated to 1.0.9. I am not really sure what the process is concerning releasing packages either. In this case it was my mistake and I will wait until after the merge to release the new package.

Comment (ricardoamador) [reply packages/buildbucket-dart/CHANGELOG.md:7]: Is there a doc on the release process we have? Also I am wondering if there is some way to test the package locally before deploying. In Java you have the ability to do this but I am not sure about dart.

**Review by keyonghan — APPROVED** [review]
LGTM

**Pull request merged**
